Updating SaaS and adding Cold Start content (#1040)
magento/merchdocs: src/marketing/product-recommendations.md

See [Recommendation Types]({% link marketing/prex-types.md %}) to learn about the available recommendation types in Magento.

## The "Cold Start" problem

The "Cold Start" problem refers to how much time a model needs to train before it can be considered high quality. In product recommendations, it translates to waiting for Adobe Sensei to train its machine learning models before deploying Recommendation units on your site.

How long that takes depends on the following variables:

- Higher traffic volume contributes to faster learning
- Some recommendation types train faster than others
- Some recommendation types do not require any training

> 

Reviewer: this seems like a fitting place for a reminder that due to complexity of this problem we suggest deploying data collection to prod first, then implementing the rest, so that models can train during development (link to our devdocs article)

{:.bs-callout-info}
If there is not enough training data collected, Magento uses [backup recommendations](#backup-recommendations) to populate your recommendation units.

| Recommendation Type | Time needed to train | Notes |
|---|---|---|
|More like this, Visual similarity | No training needed | These use catalog data only|

> |Popularity-based (Most viewed, Most purchased, Most added to cart) | Requires little training | Depends on volume of events - views are most common, then add to cart, then purchase|

Reviewer: I would say training time "varies" on this one

> |Trending | Requires three days of data to establish a popularity baseline| Trending is not about the absolute number of times something gets interacted with (via view, add to cart, purchase) but about the recent momentum of a product's popularity compared with its own popularity baseline. A product's popularity baseline is computed using a foreground set (recent popularity over 24 hours) and a background set (popularity baseline over 72 hours). If something has become much more popular recently, within 24 hours compared to its baseline popularity, then it will become a trending product. Every product has this score, and the highest one at any time comprise the set of top trending products. |

Reviewer: ```suggestion
|Trending | Requires three days of data to establish a popularity baseline| Trending is a measure of recent momentum in a product's popularity compared with its own popularity baseline. A product's trending score is computed using a foreground set (recent popularity over 24 hours) and a background set (popularity baseline over 72 hours). If an item has become much more popular within the last 24 hours as compared with its baseline popularity, then it will receive a high trending score. Every product has this score, and the highest ones at any time comprise the set of top trending products. |
```

|Viewed this, viewed that | Requires more training |Product views are decently high in volume|
|Viewed this, bought that; Bought this, bought that| Requires the most training |Purchase events are the most rare events on commerce site, especially compared to product views|

> {:.bs-callout-info}

Reviewer: I'm in favor of having this callout be part of the bulleted list above, with a link. Sometimes I feel like the callouts can disrupt the flow of information (or maybe I'm overthinking this one 🤷‍♂️ )

See [Recommendation Types]({% link marketing/prex-types.md %}) to learn about the available recommendation types in Magento.

> ## Training the machine learning models

Reviewer: Maybe "Time to train..." or "Training period for the" ? 
I want to communicate we talk about the "how long to wait?" question

> ## Training the machine learning models

Author: ```suggestion
## Time to train the machine learning models
```

> Some recommendations do not use any behavioral data and therefore do not use machine learning models to compute results. You can use those recommendations immediately.

Reviewer: ```suggestion
Some recommendation types do not use any behavioral data and therefore do not use machine learning models to compute results. You can use those recommendations immediately.
```

> Other recommendations however, require behavioral data from your shoppers to train machine learning models that build personalized recommendations. So when you can start using those types of recommendations on your storefront? It depends. The more data these models have, the more accurate and useful the recommendations will be. Collecting this data takes time. This is referred to as the "Cold Start" problem. The "Cold Start" problem is a measure of how much time a model needs to train before it can be considered high quality. In product recommendations, it translates to waiting for Adobe Sensei to train its machine learning models before deploying Recommendation units on your site. It is therefore in your best interest to deploy data collection to your production site immediately after you [install](https://devdocs.magento.com/recommendations/install-configure.html) the `magento/production-recommendations` module. While data collection is working, you can implement the [remaining tasks necessary](https://devdocs.magento.com/recommendations/implementation.html) to deploy recommendations to your storefront. Ideally, by the time you have finished testing and configuring recommendations, the machine learning models will have collected and computed enough data to build relevant recommendations thus allowing you to deploy the recommendations to your storefront.

Reviewer: ```suggestion
Other recommendation types however, require behavioral data from your shoppers to train machine learning models that build personalized recommendations. So when you can start using those types of recommendations on your storefront? It depends.  This is referred to as the "Cold Start" problem. The "Cold Start" problem is a measure of how much time a model needs to train before it can be considered high quality. In product recommendations, it translates to waiting for Adobe Sensei to train its machine learning models before deploying Recommendation units on your site. The more data these models have, the more accurate and useful the recommendations will be. Collecting this data takes time and will vary based on traffic volume. It is therefore in your best interest to deploy data collection to your production site immediately after you [install](https://devdocs.magento.com/recommendations/install-configure.html) the `magento/production-recommendations` module. While data is collected and ML models are trained, you can implement the [remaining tasks necessary](https://devdocs.magento.com/recommendations/implementation.html) to deploy recommendations to your storefront. Ideally, by the time you have finished testing and configuring recommendations, the machine learning models will have collected and computed enough data to build relevant recommendations thus allowing you to deploy the recommendations to your storefront.
```
We need an ending that inspires more confidence in the user (instead of "ideally"). we should also talk about the case when they DIDN'T wait to start data collection until the end. I would basically say something like wait as long as you can (or willing) but it should be safe to deploy within a few days.

> How long the models take to train depends on the following variables:

Reviewer: I would love to put the summary first (or at least higher), then go into detail on these points

See [Recommendation Types]({% link marketing/prex-types.md %}) to learn about the available recommendation types in Magento.

> ## Time and training for machine learning models

Author: ```suggestion
## Time and training for machine learning models {#trainmlmodels}
```

> Some recommendation types use behavioral data from your shoppers to train machine learning models that build personalized recommendations. Other recommendation types use catalog data only and do not use any behavioral data. If merchants want to start quickly, they can use the following, catalog-only recommendation types:

Reviewer 2: We usually speak directly to the merchant and not in third-person

```suggestion
Some recommendation types use behavioral data from your shoppers to train machine learning models that build personalized recommendations. Other recommendation types use catalog data only and do not use any behavioral data. If you want to start quickly, you can use the following, catalog-only recommendation types:
```

- More like this
- Visual similarity

So when can you start using recommendation types that use behavioral data? It depends. This is referred to as the "Cold Start" problem.

> The "Cold Start" problem is a measure of how much time a model needs to train before it can be considered high quality. In product recommendations, it translates to waiting for Adobe Sensei to train its machine learning models before deploying recommendation units on your site. The more data these models have, the more accurate and useful the recommendations will be. Collecting this data takes time and will vary based on traffic volume. It is therefore in your best interest to deploy data collection to your production site immediately after you [install](https://devdocs.magento.com/recommendations/install-configure.html) the `magento/production-recommendations` module.

Reviewer: ```suggestion
The "Cold Start" problem is a measure of how much time a model needs to train before it can be considered high quality. In product recommendations, it translates to waiting for Adobe Sensei to train its machine learning models before deploying recommendation units on your site. The more data these models have, the more accurate and useful the recommendations will be. Collecting this data takes time and will vary based on traffic volume. Because this data can be collected only on production, it is therefore in your best interest to deploy data collection there as early as possible. You can do this by [installing and configuring](https://devdocs.magento.com/recommendations/install-configure.html) the `magento/production-recommendations` module.
```

The following table provides some general guidance for the amount of time it takes to collect enough data for each recommendation type:

| Recommendation type | Time needed to train | Notes |
|---|---|---|

> |Popularity-based (Most viewed, Most purchased, Most added to cart) | Varies | Depends on volume of events - views are most common, then add to cart, then purchase|

Reviewer: ```suggestion
|Popularity-based (Most viewed, Most purchased, Most added to cart) | Varies | Depends on volume of events - views are most common, and therefore will train quicker; then adds to cart, then purchases|
```

|Viewed this, bought that; Bought this, bought that| Requires the most training |Purchase events are the most rare events on commerce site, especially compared to product views|
|Trending | Requires three days of data to establish a popularity baseline| Trending is a measure of recent momentum in a product's popularity compared with its own popularity baseline. A product's trending score is computed using a foreground set (recent popularity over 24 hours) and a background set (popularity baseline over 72 hours). If an item has become much more popular within the last 24 hours as compared with its baseline popularity, then it will receive a high trending score. Every product has this score, and the highest ones at any time comprise the set of top trending products. |

Other variables that can impact the time needed to train:

- Higher traffic volume contributes to faster learning
- Some recommendation types train faster than others
- Magento recomputes behavioral data every four hours. While you can technically deploy your recommendation units at that time, know that the recommendations will become more accurate the longer they are used on your site.

> While data is collected and machine learning models are trained, you can implement the [remaining tasks necessary](https://devdocs.magento.com/recommendations/implementation.html) to deploy recommendations to your storefront. By the time you have finished testing and configuring recommendations, the machine learning models will have collected and computed enough data to build relevant recommendations thus allowing you to deploy the recommendations to your storefront.

Reviewer: ```suggestion
While data is collected on production and machine learning models are trained, you can implement the [remaining tasks necessary](https://devdocs.magento.com/recommendations/implementation.html) to deploy recommendations to your storefront. By the time you have finished testing and configuring recommendations, the machine learning models will have collected and computed enough data to build relevant recommendations thus allowing you to deploy the recommendations to your storefront.
```
